NewApp/ngsa-csharp/Ngsa.Middleware/NgsaLogger/NgsaLogger.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using Microsoft.Extensions.Logging;
+
+namespace Ngsa.Middleware
+{
+    /// <summary>
+    /// Simple aspnet core middleware that logs requests to the console
+    /// </summary>
+    public class NgsaLogger : ILogger
+    {
+        private readonly ConsoleColor origColor = Console.ForegroundColor;
+        private readonly string name;
+        private readonly NgsaLoggerConfiguration config;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NgsaLogger"/> class.
+        /// </summary>
+        /// <param name="name">Logger Name</param>
+        /// <param name="config">Logger Config</param>
+        public NgsaLogger(string name, NgsaLoggerConfiguration config)
+        {
+            this.name = name;
+            this.config = config;
+        }
+
+        public IDisposable BeginScope<TState>(TState state)
+        {
+            return default;
+        }
+
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            return logLevel >= config.LogLevel;
+        }
+
+        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+        {
+            if (!IsEnabled(logLevel))
+            {
+                return;
+            }
+
```

Comment:
per @aflinchb - we should add Correlation Vector to the error log

any other fields we should add?

Reply:
I can't think of any for now - we can always add more as we learn from debugging/investigating.